### PR TITLE
business central export settings form constraction

### PR DESCRIPTION
**Pull request by DhaaraniCIT**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new `BusinessCentralExportSettingModel` for enhanced export configuration.
  - Added a new `BusinessCentralField` enum to support additional fields for business central integrations.
  - Implemented new methods in `BusinessCentralHelperService` for importing attributes and improved user feedback with toast messages.

- **Enhancements**
  - Updated the `BusinessCentralExportSettings` component with additional properties and methods to improve the onboarding process and user experience.
  - Expanded the `BusinessCentralSharedModule` with `SharedModule` import for additional shared functionalities.

- **Documentation**
  - Updated documentation to reflect new features and enhancements.

### CODE REVIEW

**Comment by ashwin1111** [comment]
@coderabbitai review

File: src/app/core/services/business-central/business-central-core/business-central-helper.service.ts [hunk 1]
```diff
@@ -11,10 +14,18 @@ export class BusinessCentralHelperService {
...
   ) { }
 
   refreshBusinessCentralDimensions(): Observable<{}> {
     return this.apiService.post(`/workspaces/${this.workspaceService.getWorkspaceId()}/Business_Central/import_attribut/`, {});
   }
+
+  importAttributes(isRefresh: boolean): void {
+    this.workspaceService.importFyleAttributes(isRefresh).subscribe();
+    this.mappingService.importBusinessCentralAttributes(isRefresh).subscribe();
+    this.toastService.displayToastMessage(ToastSeverity.SUCCESS, 'Refreshing data dimensions from Sage 300 CRE');
```
Comment (ashwin1111) [inline src/app/core/services/business-central/business-central-core/business-central-helper.service.ts:29]: content says sage300

File: src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts [hunk 2]
```diff
@@ -1,4 +1,16 @@
 import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { catchError, forkJoin, of } from 'rxjs';
+import { BusinessCentralExportSettingFormOption, BusinessCentralExportSettingGet, BusinessCentralExportSettingModel } from 'src/app/core/models/business-central/business-central-configuration/business-central-export-setting.model';
+import { ExportModuleRule, ExportSettingValidatorRule } from 'src/app/core/models/common/export-settings.model';
+import { AppName, BusinessCentralExportType, BusinessCentralField, ConfigurationCta, FyleField } from 'src/app/core/models/enum/enum.model';
+import { BusinessCentralExportSettingsService } from 'src/app/core/services/business-central/business-central-configuration/business-central-export-settings.service';
+import { HelperService } from 'src/app/core/services/common/helper.service';
+import { MappingService } from 'src/app/core/services/common/mapping.service';
+import { BusinessCentralDestinationAttributes } from '/Users/fyle/integrations/fyle-integrations-app/src/app/core/models/business-central/db/business-central-destination-attribute.model';
```
[the remark this answers is not in the document]
Comment (ashwin1111) [reply src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts:10]: +1 @DhaaraniCIT

File: src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts [hunk 3]
```diff
@@ -7,9 +19,108 @@ import { Component, OnInit } from '@angular/core';
...
+
+  creditCardAccountOptions: BusinessCentralDestinationAttributes[];
+
+  vendorOptions: BusinessCentralDestinationAttributes[];
+
+  isLoading: boolean = true;
+
+  previewImagePaths =[
+    {
+      'PURCHASE_INVOICE': 'assets/illustrations/sageIntacct/Reimbursable - Expense Report.jpg',
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts:36]: change key according to business central

File: src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts [hunk 4]
```diff
@@ -7,9 +19,108 @@ import { Component, OnInit } from '@angular/core';
...
+    const exportSettingValidatorRule: ExportSettingValidatorRule = {
+      'reimbursableExpense': ['reimbursableExportType', 'reimbursableExportGroup', 'reimbursableExportDate', 'reimbursableExpenseState'],
+      'creditCardExpense': ['cccExportType', 'cccExportGroup', 'cccExportDate', 'cccExpenseState']
+    };
+
+    const exportModuleRule: ExportModuleRule[] = [
+      {
+        'formController': 'reimbursableExportType',
+        'requiredValue': {
+          'DIRECT_COST': ['defaultReimbursableCCCAccountName', 'defaultDebitCardAccountName']
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts:96]: change key name

File: src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts [hunk 5]
```diff
@@ -7,9 +19,108 @@ import { Component, OnInit } from '@angular/core';
...
+        }
+      },
+      {
+        'formController': 'cccExportType',
+        'requiredValue': {
+          'DIRECT_COST': ['defaultCreditCardCCCAccountName', 'defaultDebitCardAccountName'],
+          'PURCHASE_INVOICE': ['defaultVendorName']
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-shared/business-central-export-settings/business-central-export-settings.component.ts:103]: same

**Comment by DhaaraniCIT** [comment]
PR comments are fixed in https://github.com/fylein/fyle-integrations-app/pull/343

**Pull request merged**